### ISSUE TITLE
Mark all frames to inherit class' direction option by default

### ISSUE THREAD

**Issue by alerque** [sile-typesetter/sile#1321]
Noticed while solving #1317...

Setting `direction=RTL` at the document level (as a class option) sets that property on the default content frame for the class. It does **not** get set on any additional frames (headers, foootnotes, folios, etc.) and it does **not** get set on new frames created by the user.

My expectation is that this class level setting would change the default for all frames created in the document, obviously with the possibility to override that with specific frame properties.

I'm not sure whether this change should be considered a breaking change or not (i.e. do we warn about it, provide a migration path, do a major revision bump, etc.) or whether this is just so expected that we should just change it already and call it good. Perhaps @khaledhosny you have an opinion on this?

**Comment by khaledhosny** [comment]
Yes, this was surprising for me. All the frame created with `\frame` command did not inherit the direction set on the document.
